aten/src/ATen/core/jit_type.h

```diff
@@ -263,7 +263,11 @@ struct SingleElementType : public Type {
   }
 
  protected:
-  SingleElementType(TypePtr elem) : Type(Kind), elem(std::move(elem)) {}
+  SingleElementType(TypePtr elem) : Type(Kind), elem(std::move(elem)) {
+    if (!this->elem) {
+      throw std::runtime_error("Can not create Generic with None type");
```

Comment:
1. `TORCH_CHECK` is preferred to throwing a regular runtime error since it has improved backtraces.
2. "Generic" is not a well-known concept for library users. Instead this error message should reference the specific `Kind`, so it looks like "can not create List with None type" or similar.

Reply:
Added `TestList.test_list_none` and updated the error message to include type name rather than `Generic`
But decided against TORCH_CHECK, because it is intended for internal, rather than for a user visible errors.